### PR TITLE
#8 Add the k8s-service-discovery as a installation component when per…

### LINKED ISSUE #8: Install k8s-service-discovery as core component when performing a setup

### PR DESCRIPTION
**Pull request by Subaro**
…forming the setup

Resolves #8

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,26 +1,41 @@
...
 
-The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
-and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres
+to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+### Added
+
```
Comment (nhinze23) [inline CHANGELOG.md:11]: delete empty lines

Comment (Subaro) [reply CHANGELOG.md]: Done.

File: app/setup/serviceDiscoveryInstallerStep_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+			k8sClient:              mockedK8sClient,
+			fileContentModificator: mockedFileModder,
+		}
+
+		// when
+		err := installer.PerformSetupStep()
+
+		// then
+		require.NoError(t, err)
+		mockedFileClient.AssertExpectations(t)
```
Comment (nhinze23) [inline app/setup/serviceDiscoveryInstallerStep_test.go:76]: you can use `mock.AssertExpectationsForObjects(t, ...)`

File: docs/operations/configuration_guide_de.md [hunk 3]
```diff
@@ -22,53 +22,56 @@ metadata:
...
 
 * YAML-Key: `dogu_operator_version`
 * Typ: `String` als Link zu der gewünschten [Dogu Operator](http://github.com/cloudogu/k8s-dogu-operator) Version
 * Notwendig Konfiguration
-* Beschreibung: Der Dogu Operator ist eine zentrale Komponente im EcoSystem und muss installiert werden. Der angegebene
-  Link zeigt auf die zu installierende Version des Dogu Operators. Der Link muss auf eine valide K8s-YAML-Ressource des
-  `k8s-dogu-operator` zeigen. Diese wird bei jeder Veröffentlichung an das Release des `k8s-dogu-operator` gehängt.
-* Beispiel: `TODO: Add first link when the first release is done`
+* Beschreibung: Der Dogu Operator ist eine zentrale Komponente im EcoSystem und muss installiert werden. Der angegebene Link zeigt auf die zu installierende Version des Dogu Operators. Der Link muss auf eine valide K8s-YAML-Ressource des `k8s-dogu-operator` zeigen. Diese wird bei jeder Veröffentlichung an das Release des `k8s-dogu-operator` gehängt.
+* Beispiel: `Thttps://github.com/cloudogu/k8s-dogu-operator/releases/download/v0.2.0/k8s-dogu-operator_0.2.0.yaml`
```
Comment (nhinze23) [inline docs/operations/configuration_guide_de.md:49]: delete `T` in Thttps

File: docs/operations/configuration_guide_en.md [hunk 4]
```diff
@@ -22,53 +22,56 @@ metadata:
...
-  created.
-
 ### log_level
 
 * YAML key: `log_level`
 * Type: one of the following values `ERROR, WARN, INFO, DEBUG`
 * Necessary configuration
-* Description: Sets the log level of the `k8s-ces-setup` and thus how accurate the log output of the application should be.
-  should be.
+* Description: Sets the log level of the `k8s-ces-setup` and thus how accurate the log output of the application should be. should be.
```
Comment (nhinze23) [inline docs/operations/configuration_guide_en.md:41]: delete should be.

File: docs/operations/configuration_guide_en.md [hunk 5]
```diff
@@ -22,53 +22,56 @@ metadata:
...
 
 * YAML key: `dogu_operator_version`
 * Type: `String` as link to the desired [Dogu Operator](http://github.com/cloudogu/k8s-dogu-operator) version
 * Necessary configuration
-* Description: The Dogu Operator is a central component in the EcoSystem and must be installed. The given
-  link points to the version of the Dogu Operator to be installed. The link must point to a valid K8s YAML resource of the
-  `k8s-dogu-operator`. This will be appended to the release of the `k8s-dogu-operator` on each release.
-* Example: `TODO: Add first link when the first release is done`
+* Description: The Dogu Operator is a central component in the EcoSystem and must be installed. The given link points to the version of the Dogu Operator to be installed. The link must point to a valid K8s YAML resource of the `k8s-dogu-operator`. This will be appended to the release of the `k8s-dogu-operator` on each release.
+* Example: `Thttps://github.com/cloudogu/k8s-dogu-operator/releases/download/v0.2.0/k8s-dogu-operator_0.2.0.yaml`
```
Comment (nhinze23) [inline docs/operations/configuration_guide_en.md:49]: s.o.

File: k8s/k8s-ces-setup.yaml [hunk 6]
```diff
@@ -135,6 +135,15 @@ rules:
       - create
       - patch
       - update
+  - apiGroups:
+      - "*"
```
Comment (nhinze23) [inline k8s/k8s-ces-setup.yaml:139]: ApiGroup from the IngressClass is networking.k8s.io

File: app/setup/serviceDiscoveryInstallerStep_test.go (hunk 2 above)
Comment (Subaro) [reply app/setup/serviceDiscoveryInstallerStep_test.go:76]: Done.

**Comment by nhinze23** [comment]
lgtm

**Pull request merged**